src/main/java/olivero/parsers/Parser.java

```diff
@@ -0,0 +1,196 @@
+package olivero.parsers;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
+
+import olivero.commands.ByeCommand;
+import olivero.commands.Command;
+import olivero.commands.CommandType;
+import olivero.commands.DeadlineCommand;
+import olivero.commands.DeleteCommand;
+import olivero.commands.EventCommand;
+import olivero.commands.FindCommand;
+import olivero.commands.ListCommand;
+import olivero.commands.MarkCommand;
+import olivero.commands.ToDoCommand;
+import olivero.commands.UnMarkCommand;
+import olivero.common.Responses;
+import olivero.exceptions.CommandParseException;
+import olivero.exceptions.UnsupportedCommandException;
+import olivero.tasks.Deadline;
+import olivero.tasks.Event;
+import olivero.tasks.ToDo;
+
+/**
+ * Parses commands from the user input and returns a {@code Command} object.
+ */
+public class Parser {
```

Comment:
```suggestion
public class InputParser {
```

Maybe renaming this to InputParser (or something similar) would better follow the naming of the other parsers, I assumed that this was the superclass of DateParser and TaskParser